tests/protocol/test_connection.py

```diff
@@ -190,6 +190,26 @@ def test_write_ulong_positive(self):
 
         assert self.connection.flush() == bytearray.fromhex("8000000000000000")
 
+    def test_read_bool_true(self):
+        self.connection.receive(bytearray.fromhex("01"))
+
+        assert self.connection.read_bool() is True
+
+    def test_write_bool_true(self):
+        self.connection.write_bool(True)
+
+        assert self.connection.flush() == bytearray.fromhex("01")
+
+    def test_read_bool_false(self):
+        self.connection.receive(bytearray.fromhex("00"))
+
+        assert self.connection.read_bool() is False
+
+    def test_write_bool_false(self):
+        self.connection.write_bool(False)
+
+        assert self.connection.flush() == bytearray.fromhex("00")
+
```

Comment:
We don't really need such basic tests, but if Kevin wants:

```suggestion
    @pytest.mark.parametrize("as_bytes,as_bool", [("01", True), ("00", False)])
    def test_read_bool(self, as_bytes: str, as_bool: bool) -> None:
        self.connection.receive(bytearray.fromhex(as_bytes))

        assert self.connection.read_bool() is as_bool

    @pytest.mark.parametrize("as_bytes,as_bool", [("01", True), ("00", False)])
    def test_write_bool(self, as_bytes: str, as_bool: bool) -> None:
        self.connection.write_bool(as_bool)

        assert self.connection.flush() == bytearray.fromhex(as_bytes)
```

P.S. I am unsure if pyright will be okay with `is as_bool`, you should check it.

Reply:
Why wouldn't it be ok? You don't need to use literals when comparing using `is` , it's fine that `as_bool` is a variable.

Reply:
I remember that mypy/flake8/pylint/something doesn't allow using `is` with variables, because of the difference between statement and expression. I don't know how it works in pyright, so just mentioned this.

Reply:
I don't know about mypy or pylint, but flake8 (at least the default one, without considering extensions) doesn't have an issue with this, and pyright doesn't either ([quick example](https://pyright-playground.decorator-factory.su/?gzip=H4sIABH9dGQC_0srys9VSM7PyUlNLsnMzyvWS0xKVsjMLcgvKlFwTszJSUzKSeXiqrBSSMrPz1GwVXBLzClO5aqE80OKSlO50qzgaqOjY3XAcrFAyZzE3KSURCuoJq7E4uJUoLEVCpnFEH1QARAbJFaBoqISxkvT0MSUBQoCAGLoEbu7AAAA))

Reply:
Good, then just ignore that P.S.

Reply:
Flake8 didn't like comparing a variable to literal boolean with `==`, it suggested changing to `is`